compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5240,3 +5240,64 @@ OMR::X86::TreeEvaluator::VMifArrayCmpEvaluator(TR::Node*, TR::CodeGenerator*)
    {
    return 0;
    }
+
+enum BinaryArithmetic_t : uint32_t
+   {
+   BinaryArithmeticInvalid,
+   BinaryArithmeticAdd,
+   BinaryArithmeticSub,
+   BinaryArithmeticMul,
+   BinaryArithmeticDiv,
+   BinaryArithmeticNum
+   };
+static const TR_X86OpCodes BinaryArithmeticOpCodes[TR::NumOMRTypes][BinaryArithmeticNum] =
+   {
+   //  Invalid,       Add,         Sub,       Mul,         Div
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // NoType
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // Int8
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // Int16
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // Int32
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // Int64
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // Float
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // Double
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // Address
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // VectorInt8
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // VectorInt16
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // VectorInt32
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // VectorInt64
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // VectorFloat
+   { BADIA32Op, ADDPDRegReg, BADIA32Op, MULPDRegReg, BADIA32Op }, // VectorDouble
+   { BADIA32Op, BADIA32Op,   BADIA32Op, BADIA32Op,   BADIA32Op }, // Aggregate
+   };
+// For ILOpCode that can be translated to single SSE/AVX instructions
```

Comment:
I personally do not think FPV is an intuitive acronym; I'd prefer to use the full term if floating point and vector operations really need be emphasized in its name. i.e. FloatingPointAndVectorBinaryArithmeticEvaluator. The function is only called from the evaluator table and hence the length of name should not bring any inconvenience to developers.